src/main/java/seedu/tasklist/model/task/Task.java

```diff
@@ -109,12 +122,25 @@ public void setPriority(Priority priority){
         this.priority = priority;
     }
     
+    public boolean isFloating(){
+    	return endTime.isMissing()&&startTime.isMissing();
+    }
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof ReadOnlyTask // instanceof handles nulls
                 && this.isSameStateAs((ReadOnlyTask) other));
     }
+    
+    public boolean isOverDue(){
+    	if(!isFloating()){
+      int hours = endTime.endtime.get(Calendar.HOUR_OF_DAY);
+      Calendar today = Calendar.getInstance();
+      today.set(Calendar.HOUR_OF_DAY, hours);
+      return endTime.endtime.before(today);
```

Comment:
return endTime.endtime.getTimeInMillis() < System.currentTimeMillis();

can replace the whole block and functions properly